model/src/main/java/org/projectnessie/api/http/HttpTreeApi.java

```diff
@@ -571,6 +572,85 @@ void mergeRefIntoBranch(
           Merge merge)
       throws NessieNotFoundException, NessieConflictException;
 
+  /**
+   * Commit multiple operations against the given branch expecting that branch to have the given
+   * hash as its latest commit. The hash in the successful response contains the hash of the commit
+   * that contains the operations of the invocation.
+   *
+   * @param branchName Branch to change, defaults to default branch.
+   * @param hash Expected hash of branch.
+   * @param operations {@link Operations} to apply
+   * @return updated {@link Branch} objects with the hash of the new HEAD
+   * @throws NessieNotFoundException if {@code branchName} could not be found
+   * @throws NessieConflictException if the operations could not be applied to some conflict, which
+   *     is either caused by a conflicting commit or concurrent commits.
+   */
+  @POST
+  @Path("branch/{branchName}/commit")
+  @Consumes(ApiVersions.APPLICATION_NESSIE_GLOBAL_STATE)
+  @Produces(ApiVersions.APPLICATION_NESSIE_GLOBAL_STATE)
+  @Operation(
+      summary =
+          "Commit multiple operations against the given branch expecting that branch to have "
+              + "the given hash as its latest commit. The hash in the successful response contains the hash of the "
+              + "commit that contains the operations of the invocation.")
+  @APIResponses({
+    @APIResponse(
+        responseCode = "200",
+        description = "Updated successfully.",
+        content = {
+          @Content(
+              mediaType = ApiVersions.APPLICATION_NESSIE_GLOBAL_STATE,
+              examples = {@ExampleObject(ref = "refObj")},
+              schema = @Schema(implementation = Branch.class))
+        }),
+    @APIResponse(responseCode = "400", description = "Invalid input, ref/hash name not valid"),
+    @APIResponse(responseCode = "401", description = "Invalid credentials provided"),
+    @APIResponse(
+        responseCode = "403",
+        description = "Not allowed to view the given reference or perform commits"),
+    @APIResponse(responseCode = "404", description = "Provided ref doesn't exists"),
+    @APIResponse(responseCode = "409", description = "Update conflict")
+  })
+  Branch commitMultipleOperationsWithGlobalState(
```

Comment:
Could we handle this as a `default` method and redirect to `commitMultipleOperations`? It looks like `Operations` is able to contain both per-ref and global operations at the same time... Are we going to have different processing logic in this API method?

Reply:
Yes. those will be a little bit different. `commitMultipleOperations` only takes contents using the current (aka Nessie API 0.9) semantics w/o global state. `commitMultipleOperationsWithGlobalState` expects Nessie API v1 semantics w/ per-reference state in `Put` and global-state via `PutGlobal`.

It's a bit tricky to untangle the object hierarchy here for several reasons:
* type-parameters in {@code StoreWorker} and everything that depends on it. adding another type-parameter adds a ton of noise.
* backwards-compatibility of the API makes it necessary to rely on the most common-type (`Contents`)
* content-types for delta-lake + sql-view do not have a global state, therefore do not extend `OnReferenceState`, which expects the global state type

Reply:
Right, but if we simply validate that in v0.9 API we do not have `PutGlobal` operations (and error out if we do), will the subsequent processing not be the same?

Reply:
same comment as before, why does the end user care about the global part of some contents.

Reply:
additionally, I don't think we should make any allowances for old pre-1.0 apis. We should remove them from 1.0 if at all possible. Clean break from pre-1.0